File: android/Lib/src/main/kotlin/com/darakeon/dfm/lib/api/entities/Environment.kt
package com.darakeon.dfm.lib.api.entities

data class Environment(
	val theme: ThemeEnum,
	val language: String,
	val tfaForgottenWarning: Boolean
) {
	constructor(language: String)
		: this(Theme.None, language)

	constructor(theme: Theme)
		: this(theme, "")

	constructor(tfaForgottenWarning: Boolean)
		: this(Theme.None, "", tfaForgottenWarning)

	constructor(theme: Theme, language: String)
		: this(theme, language, false)

	constructor(theme: Theme, language: String, tfaForgottenWarning: Boolean)
		: this(ThemeEnum(theme.code), language, tfaForgottenWarning)
}

File: api/API/Models/Environment.cs
using System;
using DFM.Generic;

namespace DFM.API.Models
{
	public class Environment(
		Theme theme,
		string language,
		Boolean tfaForgottenWarning
	)
	{
		public Theme Theme { get; } = theme;
		public String Language { get; } = language;
		public Boolean TFAForgottenWarning { get; } = tfaForgottenWarning;
	}
}
